src/libs/Notification/PushNotification/shouldShowPushNotification.ts

```diff
@@ -1,30 +1,31 @@
-import _ from 'underscore';
+import {PushPayload} from '@ua/react-native-airship';
 import Log from '@libs/Log';
 import * as ReportActionUtils from '@libs/ReportActionsUtils';
 import * as Report from '@userActions/Report';
+import {NotificationData} from './NotificationType';
 
 /**
  * Returns whether the given Airship notification should be shown depending on the current state of the app
- * @param {PushPayload} pushPayload
- * @returns {Boolean}
  */
-export default function shouldShowPushNotification(pushPayload) {
+export default function shouldShowPushNotification(pushPayload: PushPayload): boolean {
     Log.info('[PushNotification] push notification received', false, {pushPayload});
 
-    let pushData = pushPayload.extras.payload;
+    let payload = pushPayload.extras.payload;
```

Comment:
```suggestion
    let payload = pushPayload.extras.payload as NotificationData;
```

Reply:
Same here, I think is not correct to assert to NotificationData at this point, since payload can be a string and a few lines below we parse it to JSON.